### PR TITLE
Update code scanning alert

### PR DESCRIPTION
**Pull request by adrienpessu**
This pull request introduces a new feature for [updating code scanning alerts in a GitHub repository](https://docs.github.com/en/rest/code-scanning/code-scanning?apiVersion=2022-11-28#update-a-code-scanning-alert). The most important changes include adding a new module for handling updates, defining a new enum for alert states, and creating a test to verify the update functionality.

### New Feature: Update Code Scanning Alerts

* [`src/api/code_scannings.rs`](diffhunk://#diff-1b917f448d02f4156ee44813b4eaf6473910424ab7df65240bd2b86e47ad2d20R5): Added the `update` module and the `update` method to the `CodeScanningHandler` to handle updating code scanning alerts. [[1]](diffhunk://#diff-1b917f448d02f4156ee44813b4eaf6473910424ab7df65240bd2b86e47ad2d20R5) [[2]](diffhunk://#diff-1b917f448d02f4156ee44813b4eaf6473910424ab7df65240bd2b86e47ad2d20R44-R64)
* [`src/api/code_scannings/update.rs`](diffhunk://#diff-e9c5b1ad72bef7eb44fb3c9e0fa008dfe1f2b23cea5b2c8428af9006aa6de8efR1-R54): Implemented the `UpdateCodeScanningBuilder` struct, which includes methods for setting the state, dismissed reason, and dismissed comment of a code scanning alert, and a `send` method to execute the update.
* [`src/params.rs`](diffhunk://#diff-96db4c2e87979d88bcea716895d0d011b21c2243b7c1e2175e73a25259e43578R3-R11): Added the `AlertState` enum to represent the state of a code scanning alert, with possible values `Dismissed` and `Open`.
* [`tests/code_scanning_alert_update_test.rs`](diffhunk://#diff-11ab8640b503a8fb13dc8dd109a506914be6f45bf914a21d51b178b1de982c71R1-R65): Created a test to verify that the `update` method correctly updates a code scanning alert and handles a successful response.

### CODE REVIEW

File: src/api/code_scannings.rs [hunk 1]
```diff
@@ -40,4 +41,25 @@ impl<'octo> CodeScanningHandler<'octo> {
     pub fn list(&self) -> list::ListCodeScanningsBuilder<'_, '_> {
         list::ListCodeScanningsBuilder::new(self)
     }
+
+    /// Update an issue in the repository.
```
Comment (XAMPPRocky) [inline src/api/code_scannings.rs:45]: Shouldn't this be updating a code scanning?

File: src/api/code_scannings/update.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    pub fn dismissed_comment(mut self, dismissed_comment: impl Into<String>) -> Self {
+        self.dismissed_comment = Some(dismissed_comment.into());
+        self
+    }
+
+    pub async fn send(self) -> Result<models::code_scannings::CodeScanningAlert> {
+        let route = format!(
+            "/repos/{owner}/{repo}/code-scanning/alerts/{code_scanning}",
+            owner = self.handler.owner,
+            repo = self.handler.repo.as_ref().unwrap_or(&"".to_string()),
```
Comment (XAMPPRocky) [inline src/api/code_scannings/update.rs:48]: Shouldn't this be a required argument?

File: src/api/code_scannings.rs (hunk 1 above)
Comment (adrienpessu) [reply src/api/code_scannings.rs]: Good call

File: src/api/code_scannings/update.rs (hunk 2 above)
Comment (adrienpessu) [reply src/api/code_scannings/update.rs]: You're right. I changed to `.expect()` but I can make the repo a `String` instead of an `Option` if it's cleaner

**Pull request merged**